core/tests/services_sources/ATestFactory.ts

```diff
@@ -1,27 +0,0 @@
-// Copyright 2015 The Oppia Authors. All Rights Reserved.
```

Comment:
Why was this file deleted? We were going to exclude these, not delete these.

Reply:
Sorry I got it wrong. Do I supposed to exclude these files from where? From the regex? I didn't quite understand the difference between exclude and delete it.

Reply:
"delete" means remove the file from the codebase.

"exclude" means keep the file in the codebase, but do not count it towards coverage and don't run tests with it.

Reply:
I was already doing it in the regex. I was ignoring these files to not be tested.

Reply:
I don't see the exclusion of services_sources in the regex. Can you point me to the line of code that does it?

Also please note that the CI tests are failing on this PR. Please fix that before requesting a follow-up review.

Reply:
Fixed it!

The exclusion of these files are in the `combined-tests.spec.ts` regex using negative lookbehind:
`(?<!ATestFactory\.ts)(?<!BTestService\.ts)(?<!CTest\.service\.ts)(?<!DTest\.service\.ts)(?<!ETestFactory\.ts)(?<!F\.directive\.ts)`

Reply:
Hi @marianazangrossi, two things:

- The exclusion rule should be based on "services_sources", which I don't see in your PR. It should exclude that entire folder, because those files do not contribute to the Karma testing.
- Please never force push PRs. We generally close PRs that have been force pushed because it makes it hard to review the diffs. Please follow [these instructions](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change) for all future PRs.

Thanks!

Reply:
Hey @seanlip, I've fixed the exclusion of `services_sources` folder but I can't push it, npm is showing a vulnerability related to angular.js

![Screenshot from 2019-11-30 21-04-54](https://user-images.githubusercontent.com/34922478/69907512-13c66980-13b5-11ea-9902-b15ceba6a9f7.png)

By the way, I tried to exclude the folder using the `exclude` property from `karma.conf.ts` but it didn't work, since the files were being loaded by `require.context` function. So the solution was to exclude the entire folder using the regex.

Reply:
Ah, thanks @marianazangrossi. I need to defer to @kevinlee12 @ankita240796 on the error you're finding here...

Reply:
Thanks for flagging this, we're aware about this and @ankita240796 is working on it. I just worked on a PR that will unblock this immediate issue and will let you know when that is merged.

Reply:
Hi @marianazangrossi, @kevinlee12's PR has gone in. Could you please update from develop and try pushing again? Let us know if you have any problems.

Thanks!

Reply:
Sure! Done!